client/blocks/app-banner/index.jsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import { getActionLog } from 'state/ui/action-log/selectors';
+import {
+	getAppBannerData,
+	getPageType,
+	isDismissed,
+	getNewDismissTimes,
+	ALLOWED_PAGE_TYPES
+} from './utils';
+import {
+	findLast,
+	get,
+	identity,
+	includes,
+	noop
+} from 'lodash';
+import { isMobile } from 'lib/viewport';
+import { getUserSetting, isNotificationsOpen } from 'state/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { ROUTE_SET } from 'state/action-types';
+import { saveUserSettings } from 'state/user-settings/actions';
+
+class AppBanner extends Component {
```

Comment:
Thanks! Updated dismiss logic to match that of `DismissibleCard` in c6910501c8a4785fad17b0aa07659b63cea82c71.